src/storage/src/source/postgres/snapshot.rs

```diff
@@ -560,39 +589,53 @@ pub(crate) fn render<G: Scope<Timestamp = MzOffset>>(
     (
         snapshot_updates,
         rewinds,
+        slot_ready,
         stats_stream,
         errors,
         button.press_on_drop(),
     )
 }
 
 /// Starts a read-only transaction on the SQL session of `client` at a consistent LSN point by
-/// creating a temporary replication slot. Returns a snapshot identifier that can be imported in
+/// creating a replication slot. Returns a snapshot identifier that can be imported in
 /// other SQL session and the LSN of the consistent point.
-async fn export_snapshot(client: &Client) -> Result<(String, MzOffset), TransientError> {
+async fn export_snapshot(
+    client: &Client,
+    slot: &str,
+    temporary: bool,
+) -> Result<(String, MzOffset), TransientError> {
     client
         .simple_query("BEGIN READ ONLY ISOLATION LEVEL REPEATABLE READ;")
         .await?;
-    // A temporary replication slot is the only way to get the tx in a consistent LSN point
-    let slot = format!("mzsnapshot_{}", uuid::Uuid::new_v4()).replace('-', "");
-    let query =
-        format!("CREATE_REPLICATION_SLOT {slot:?} TEMPORARY LOGICAL \"pgoutput\" USE_SNAPSHOT");
-    let row = simple_query_opt(client, &query).await?.unwrap();
-    let consistent_point: PgLsn = row.get("consistent_point").unwrap().parse().unwrap();
 
-    let row = simple_query_opt(client, "SELECT pg_export_snapshot();")
-        .await?
-        .unwrap();
-    let snapshot = row.get("pg_export_snapshot").unwrap().to_owned();
+    // Note: Using unchecked here is okay because we're using it in a SQL query.
+    let slot = Ident::new_unchecked(slot).to_ast_string();
```

Comment:
why did we need to introduce `Ident::new_unchecked` here?

Reply:
It was being done before, check the `ensure_replication_slot` function. This is making sure "weird" slot names are escaped appropriately